### DISCUSSION TITLE
MasterDetailRelation with GUID

### DISCUSSION THREAD

**Discussion by dfaruque** [serenity-is/Serenity#6325] [Help] [answered]
In the UPDATE mode it works as expected. 
but in the CREATE mode, if we add some detail then it says "Master Id field is required!"
the master id field is 
```
        [Identity, IdProperty] //also tried [AutoIncrement, PrimaryKey]
        public Guid? Id
        {
            get => fields.Id[this];
            set => fields.Id[this] = value;
        }
```

any mistake?

**Comment by minhhungit** [comment, marked as answer]
I also have a case that use GUID too, and I have to generate that ID guid in Repository (.net framework version)
```csharp
protected override void ExecuteSave()
{
    if (IsCreate && this.Row.Id == null)    
    {
        this.Row.Id = Guid.NewGuid();
    }
    base.ExecuteSave();
}
```

**Reply by dfaruque** [reply]
Thanks. Your trick is working.

looks like the GUID generated at the database end is not returned after save. so the EntityId is `null` 
`{"EntityId":null,"Error":null,"CustomData":null}`

**Reply by flashyjunior** [reply]
Thanks @minhhungit, been spending days trying to resolve this same issue
